metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1009,7 +999,7 @@ private QuorumController(LogContext logContext,
             snapshotRegistry, sessionTimeoutNs, replicaPlacer);
         this.featureControl = new FeatureControlManager(supportedFeatures, snapshotRegistry);
         this.producerIdControlManager = new ProducerIdControlManager(clusterControl, snapshotRegistry);
-        this.snapshotGeneratorManager = new SnapshotGeneratorManager(snapshotWriterBuilder);
+        this.snapshotGeneratorManager = new SnapshotGeneratorManager(raftClient::createSnapshot);
```

Comment:
Passing through the function is a tad odd. We actually could just use the implicit reference to `raftClient`. Was this done for testing or something?